### ISSUE TITLE
Error message hard to understand when sync client is blocked

### ISSUE THREAD

**Issue by schiessle** [nextcloud/terms_of_service#959]
As long as the user didn't accepted the ToS, access to the files is blocked by the sync client. 

The user gets a sabredav error from the desktop client, something like "file not available". A normal user will never figure out what's the problem and how to solve it. It would be great if we could show a error which tells the user that they have to go to the web interface and accept the ToS to sync their files again.

Maybe the ToS app could through a exception with this error message which than gets forwarded to the clients?

**Comment by tobiasKaminsky** [comment]
Maybe we can c&p the mechanism from remote wipe?
There the entire webdav is blocked.

**Comment by nickvergessen** [comment]
Sorry, but `isReadable` is overwritten to return false. Should be good enough!?
I also can not see any code that would block to-be-wiped tokens specifically.

**Comment by tobiasKaminsky** [comment]
Remote wipe is returning 401 UNAUTHORIZED when remote wipe is enabled for that token.

**Comment by nickvergessen** [comment]
Yeah but we can't/shouldn't do that for ToS, as 401 means throw away your credentials?

**Comment by tobiasKaminsky** [comment]
I meant this as a way to check how this is done.
For ToS I would use another code, e.g. forbidden?
If then exception class is TOS, client can show correct TOS directly.

**Comment by nickvergessen** [comment]
Yes, but that is what I tried to say earlier, we only change/overwrite a boolean to false:
https://github.com/nextcloud/terms_of_service/issues/959#issuecomment-2438072085

The wrapper system we have does not allow us passing a reason, app name or anything alike 🙈
And throwing in a boolean check would be breaking behaviour and require fixing many things in many apps that access files

**Comment by nickvergessen** [comment]
Same is done by access control:
https://github.com/nextcloud/files_accesscontrol/blob/main/lib/StorageWrapper.php#L85

**Comment by tobiasKaminsky** [comment]
Hm. Maybe a short call? 
As I do not want to discuss what is working now, but how we can enhance it :)
